R/zchunk_LB164.ag_Costs_USA_C_2005_irr.R

```diff
@@ -36,43 +42,130 @@ module_aglu_LB164.ag_Costs_USA_C_2005_irr_DISABLED <- function(command, ...) {
     USDA_cost_data <- get_data(all_data, "aglu/USDA_cost_data")
     L100.LDS_ag_HA_ha <- get_data(all_data, "L100.LDS_ag_HA_ha")
     L133.ag_Cost_75USDkg_C <- get_data(all_data, "L133.ag_Cost_75USDkg_C")
-    L161.ag_irrHA_frac_R_C_GLU <- get_data(all_data, "L161.ag_irrHA_frac_R_C_GLU")
+    L161.ag_irrHA_frac_R_C_GLU <- get_data(all_data, "temp-data-inject/L161.ag_irrHA_frac_R_C_GLU")
+
+
+    # Perform computations
+    # old comment: The method here is to start from the cost file that has already been processed, and
+    #              to deduct costs of purchased irrigation water from each cost estimate. This is less
+    #              repetitive than following all of the same steps in that prior file, but with irrigation
+    #              water mapped elsewhere.
+
+    # Step 1: Compute the share of total variable costs from purchased irrigation water.
+    # CostFrac = IrrCost / TotCost
+    USDA_cost_data %>%
+      gather(year, cost, -Crop, -Item, -Unit) %>%
+      mutate(year = as.integer(year)) %>%
+      filter(Item == "Purchased irrigation water" | Item == "Total operating costs",
+             year %in% MODEL_COST_YEARS) %>%
+      spread(Item, cost) %>%
+      rename(IrrCost = `Purchased irrigation water`,
+             TotCost = `Total operating costs`) %>%
+      mutate(CostFrac = IrrCost / TotCost) %>%
+      # filter out any crops with IrrCost in all years
+      group_by(Crop) %>%
+      filter(!all(is.na(IrrCost))) %>%
+      select(-Unit, -IrrCost, -TotCost) %>%
+      # calculate average Cost fraction for each crop, over MODEL_COST_YEARS
+      mutate(waterCostFrac = mean(CostFrac, na.rm = T)) %>%
+      ungroup %>%
+      left_join_error_no_match(USDA_crops, by = c("Crop" = "USDA_crop")) ->
+      L164.waterCostFrac_Cusda
+
+    # Step 2: Use L100 LDS harvested area data to weight the waterCostFrac for each USDA
+    # crop when aggregating to GCAM commodities that are represented in USDA cost spreadsheets.
+    # Use the aggregated weighted waterCostFrac and weights to compute weighted average water
+    # cost fraction by GCAM commodities that are represented in USDA cost spreadsheets.
+    L100.LDS_ag_HA_ha %>%
+      filter(GTAP_crop %in% USDA_crops$GTAP_crop &
+             iso == "usa") %>%
+      group_by(iso, GTAP_crop) %>%
+      summarise(weight = sum(value)) %>%
+      ungroup %>%
+      select(-iso) %>%
+      left_join_error_no_match(L164.waterCostFrac_Cusda, ., by = "GTAP_crop") %>%
+      mutate(waterCostFrac_wt = waterCostFrac * weight) %>%
+      group_by(GCAM_commodity) %>%
+      summarise(waterCostFrac_wt = sum(waterCostFrac_wt),
+                weight = sum(weight)) %>%
+      ungroup %>%
+      # compute weighted average water cost fraction by GCAM commodities that are represented in USDA cost spreadsheets.
+      mutate(waterCostFrac = waterCostFrac_wt / weight) ->
```

Comment:
I don't believe so.
A weighted average of some set of quantities x_i is generally `sum(w_i * x_i) / sum(w_i).`

Your definition only holds for a weighted average when all the weights already sum to 1 (which they do not here).  Essentially, the calculations in the code now are introducing that fractional nature.
To use `weighted.mean`, I believe we would have to go through and adjust the weights to be a fraction out of total. Not difficult, but also the same number of operations (I think) as the code here. 


In other words, with the approach in the code now, we get terms like ` (w1*x1 + w2*x2) / (w1 + w2)`, which is what we want.

(We have these desired terms because `waterCostFrac_wt = waterCostFrac * weight`)